packages/react-dom/src/__tests__/validateDOMNesting-test.internal.js

```diff
@@ -9,176 +9,67 @@
 
 'use strict';
 
-var validateDOMNesting;
+var React;
+var ReactDOM;
 
-// https://html.spec.whatwg.org/multipage/syntax.html#special
-var specialTags = [
-  'address',
-  'applet',
-  'area',
-  'article',
-  'aside',
-  'base',
-  'basefont',
-  'bgsound',
-  'blockquote',
-  'body',
-  'br',
-  'button',
-  'caption',
-  'center',
-  'col',
-  'colgroup',
-  'dd',
-  'details',
-  'dir',
-  'div',
-  'dl',
-  'dt',
-  'embed',
-  'fieldset',
-  'figcaption',
-  'figure',
-  'footer',
-  'form',
-  'frame',
-  'frameset',
-  'h1',
-  'h2',
-  'h3',
-  'h4',
-  'h5',
-  'h6',
-  'head',
-  'header',
-  'hgroup',
-  'hr',
-  'html',
-  'iframe',
-  'img',
-  'input',
-  'isindex',
-  'li',
-  'link',
-  'listing',
-  'main',
-  'marquee',
-  'menu',
-  'menuitem',
-  'meta',
-  'nav',
-  'noembed',
-  'noframes',
-  'noscript',
-  'object',
-  'ol',
-  'p',
-  'param',
-  'plaintext',
-  'pre',
-  'script',
-  'section',
-  'select',
-  'source',
-  'style',
-  'summary',
-  'table',
-  'tbody',
-  'td',
-  'template',
-  'textarea',
-  'tfoot',
-  'th',
-  'thead',
-  'title',
-  'tr',
-  'track',
-  'ul',
-  'wbr',
-  'xmp',
-];
+function expectInvalidNestingWarning(shouldWarn, tags) {
+  // TODO: I get the error <spyOn> : error has already been spied upon although
+  // i am resetting the jest modules before every assertion
```

Comment:
Yeah. So, `jest.resetModules()` only resets requires, forcing next `require()` to load fresh versions. But it doesn't actually reset global objects like `console`. 

I think it's fine to just put `spyOnDev(console, 'error')` in the beginning of every test case. That's pretty explicit.